src/query_graph/graph_path.rs

```diff
@@ -438,6 +455,12 @@ impl std::fmt::Debug for SimultaneousPaths {
     }
 }
 
+impl std::fmt::Display for SimultaneousPaths {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        self.fmt_indented(&mut IndentedFormatter::new(f))
+    }
+}
+
```

Comment:
I'd just have the fmt_indented straight up in here, I don't think it needs an extra Impl block.

I also feel a bit strange about matching on sliced elements. A nicer pattern would be to do a `.split_first()`.

```suggestion
impl std::fmt::Display for SimultaneousPaths {
    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
        let mut state = IndentedFormatter::new(f);

        if let Some((first, elements)) = self.0.split_first() {
            state.write_fmt(format_args!("{{ {first} }}"))?;
            state.write("{")?;
            write_indented_lines(&mut state, elements, |f, elem| f.write(elem))?;
            state.write("}")
        } else {
            state.write("<no path>")
        }
    }
}
```

Reply:
I agree with Iryna that we should just move that logic to the display trait as even if `fmt_indented` was used elsewhere in JS codebase -> if display is just referencing the `fmt_indented` in all other places anyway we could now just rely on the display to do the same thing.

Reply:
It's an anti-pattern in Rust to have formatting done in a non-Display or non-Debug implementations. The Rust implementation should be using those whenever a formatted version of a struct is needed, and not a one-off method. It's an expectation of the ecosystem.